drivers/lm75/lm75.c

```diff
@@ -0,0 +1,281 @@
+        return LM75_ERROR_I2C;
+    }
+
+    i2c_release(I2C_BUS);
+    return LM75_SUCCESS;
+}
+
+int lm75_get_os_temp(lm75_t *dev, int *temperature) {
+
+    uint16_t temp;
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+    /* read the temperature register */
+    if (i2c_read_regs(I2C_BUS, I2C_ADDR, LM75_TOS_REG, &temp, sizeof(temp), 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+
+    i2c_release(I2C_BUS);
+    *temperature = (int16_t)htons(temp) * dev->lm75_params.lm75_res.os_res >> dev->lm75_params.lm75_res.os_bits;
+    return LM75_SUCCESS;
+}
+
+int lm75_get_hyst_temp(lm75_t *dev, int *temperature) {
+
+    uint16_t temp;
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    /* read the temperature register */
+    if (i2c_read_regs(I2C_BUS, I2C_ADDR, LM75_THYST_REG, &temp, sizeof(temp), 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+
+    i2c_release(I2C_BUS);
+    *temperature = (int16_t)htons(temp) * dev->lm75_params.lm75_res.os_res >> dev->lm75_params.lm75_res.os_bits;
+    return LM75_SUCCESS;
+}
+
+int lm75_shutdown(lm75_t *dev) {
+
+    if (dev->lm75_params.shutdown_mode == SHUTDOWN_MODE) {
+        LOG_INFO("device already in shutdown mode\n");
+        return LM75_ERROR;
+    }
+
+    if(i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    uint8_t config = 0;
+
+    if(i2c_read_reg(I2C_BUS, I2C_ADDR, LM75_CONF_REG, &config, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR;
+    }
+
+    config |= 0b00000001;
+
+    if(i2c_write_reg(I2C_BUS, I2C_ADDR, LM75_CONF_REG, config, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR;
+    }
+
+    i2c_release(I2C_BUS);
+    dev->lm75_params.shutdown_mode = SHUTDOWN_MODE;
+    return LM75_SUCCESS;
+}
+
+int lm75_wake(lm75_t *dev) {
+
+     if (dev->lm75_params.shutdown_mode == NORMAL_MODE) {
+        LOG_INFO("device is already awake\n");
+        return LM75_ERROR;
+    }
+
+    if(i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    uint8_t config = 0;
+
+    if(i2c_read_reg(I2C_BUS, I2C_ADDR, LM75_CONF_REG, &config, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR;
+    }
+
+    config &= 0b11111110;
+
+    if(i2c_write_reg(I2C_BUS, I2C_ADDR, LM75_CONF_REG, config, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR;
+    }
+
+    i2c_release(I2C_BUS);
+    dev->lm75_params.shutdown_mode = NORMAL_MODE;
+    return LM75_SUCCESS;
+}
+
+/* Performs a single temperature conversion from shutdown mode and goes back into shutdown afterwards */
+int tmp1075_one_shot(lm75_t *dev) {
+
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    #if !MODULE_TMP1075
+        i2c_release(I2C_BUS);
+        LOG_ERROR("Device incompatible with the one shot conversion function\n");
+        return LM75_ERROR;
+
+    #else
+
```

Comment:
shouldn't we check the device type still?

Reply:
I don't quite understand what you mean here. You mean I should remove the device check or make it a bit more restrictive?
Checking the device type in this function is currently redundant as the function is only ran if the device type was already checked as a TMP1075, but it might be useful in the future, maybe.

Reply:
If `dev->res` points to the device configuration type, we could instead do

    if (dev->res != tmp1075_res)

and have the possibility to mix & match different sensor types on the same board.